### PR TITLE
optimizing time complexity of split field validator

### PR DESCRIPTION
**Pull request by Andrewq11**
## Changelogs

- Updates to the `split` field validator in the `Benchmark` model which significantly improve the validators time complexity (thank you for the suggestions and analysis, @jstlaurent and @cwognum)

---

_Checklist:_

- [x] _Was this PR discussed in an issue? It is recommended to first discuss a new feature into a GitHub issue before opening a PR._ https://github.com/polaris-hub/polaris/issues/127
- [x] _Add tests to cover the fixed bug(s) or the newly introduced feature(s) (if appropriate)._
- [x] _Update the API documentation if a new function is added, or an existing one is deleted._
- [x] _Write concise and explanatory changelogs above._
- [x] _If possible, assign one of the following labels to the PR: `feature`, `fix`, `chore`, `documentation` or `test` (or ask a maintainer to do it for you)._

---

_discussion related to that PR_

### CODE REVIEW

File: polaris/benchmark/_base.py [hunk 1]
```diff
@@ -180,28 +181,33 @@ def _validate_split(cls, v, info: ValidationInfo):
...
+        if len(train_idx_set) != len(train_idx_list):
             raise InvalidBenchmarkError("The training set contains duplicate indices")
-        if len(set(test_indices)) != len(test_indices):
+        if len(test_idx_set) != len(test_idx_list):
             raise InvalidBenchmarkError("The test set contains duplicate indices")
 
         # All indices are valid given the dataset
         if info.data["dataset"] is not None:
-            if any(i < 0 or i >= len(info.data["dataset"]) for i in train_indices + test_indices):
+            max = len(info.data["dataset"])
```
Comment (cwognum) [inline polaris/benchmark/_base.py:207]: **Suggestion:** I would rename this variable.

```suggestion
            max_i = len(info.data["dataset"])
```


Naming a variable `max` overwrites the built-in `max()` function. E.g.:

```
>>> max(1, 3)
3
>>> max = 3
>>> max(1, 3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'int' object is not callable
```

**Review by cwognum — APPROVED** [review]
Thanks @Andrewq11 ! 

Did you try splitting a large dataset again? How long did that take?

File: polaris/benchmark/_base.py (hunk 1 above)
Comment (Andrewq11) [reply polaris/benchmark/_base.py:207]: Good catch!

**Comment by Andrewq11** [comment]
> Did you try splitting a large dataset again? How long did that take?

@cwognum I did try splitting a large dataset! I believe 200,000,000 rows took ~20 seconds and 100,000,000 took ~8 seconds. Much, much better.

It would be great to have more logging for when these validations are happening too. Just something to let the user know that we are running validations which can take some time (10-20 seconds of "hanging" could be confusing for a user). What do you think? Happy to create an issue.

**Comment by cwognum** [comment]
Yeah, that makes sense to me. Not super high priority because I don't expect such large datasets any time soon, but improving the logging is an important, low hanging fruit to improve the UX!

**Pull request merged**